src/cluster/registration.go

```diff
@@ -51,20 +53,33 @@ func (c DeviceCluster) HandleDiscoveredDevice(newDevice model.NewDevice) error {
 		return err
 	}
 
-	if len(results) > 0 {
-		return nil
-	}
-
-	completeDevice, err := c.DeviceRepository.Create(newDevice)
-	if err != nil {
-		return err
+	discoveredDevice := new(model.Device)
+	switch len(results) {
+	case 0:
+		discoveredDevice, err = c.DeviceRepository.Create(newDevice)
+		if err != nil {
+			return err
+		}
+	case 1:
+		discoveredDevice := device.FromNewDevice(newDevice)
+		discoveredDevice.Active = true
+		err := c.DeviceRepository.Update(device.UpdateDeviceFromDevice(&discoveredDevice))
+		if err != nil {
+			return err
+		}
+	default:
+		return errors.New("multiple results returned for 1 mac address")
 	}
 
-	c.logger.Debug(fmt.Sprintf("registered mac address [%s] with id [%s]", completeDevice.MAC, completeDevice.ID))
+	c.logger.Debug(
+		fmt.Sprintf("registered mac address [%s] with id [%s] at [%s]:[%s]",
+			discoveredDevice.MAC,
+			discoveredDevice.ID,
+			newDevice.Host,
+			strconv.Itoa(newDevice.Port)))
 
-	deviceWrapper := device.NewDeviceWrapper(*completeDevice)
-	err = deviceWrapper.RunHealthCheck(c.DeviceClient)
-	if err != nil {
+	// Immediately run health check
+	if err := device.NewDeviceWrapper(*discoveredDevice).RunHealthCheck(c.DeviceClient); err != nil {
```

Comment:
I think this makes this function a bit multi-purpose. This should prolly be run right after the `HandleDiscoveredDevice` function is run